projects/distributed-tracing/src/shared/graphql/model/schema/specifier/specification.ts

```diff
@@ -10,3 +10,8 @@ export interface Specification {
   asGraphQlOrderByFragment(): GraphQlSortWithoutDirection;
   extractFromServerData(resultContainer: Dictionary<unknown>): unknown;
 }
+
+export interface SpecificationResult {
+  alias: string;
```

Comment:
Can we get a type on the data? at a minimum, `T`? It seems like it should reflect the potential presence of a units field though.

Reply:
Every specification result can be different. We may be able to type it here, but in `buildSpecificationResultWithUnits` I don't think we can pass all the types, so i may have to pass unknown only. So I am not sure what we will achieve.

Reply:
the value might be different (although the specific shape depends on the caller), but what is under the purview of the new code is the presents of a unit field. For example, to match the metadata logic we could do:
```
export interface SpecificationResult<T> {
  alias: string;
  data: T extends MetricAggregation ? T & { units : string} : T;
}
```